### PR TITLE
Fixed APPolicy CRD

### PR DESCRIPTION
**Pull request by fabriziofiorucci**
### Proposed changes

- Added the missing `maskValueInLogs` to cookies for app-protect
- Added the missing `useXmlResponsePage` to `xml-profiles` for app-protect
- Changed several types from string to integer in `spec.policy.xml-profiles[0].defenseAttributes`
- Added missing enums in `blocking-settings.properties.evasions.items.properties.description.enum`

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [X] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have checked that all unit tests pass after adding my changes
- [X] I have updated necessary documentation
- [X] I have rebased my branch onto main
- [X] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

**Comment by schweits** [comment]
Not sure if you can change anything, but according to the NAP schema both ints and strings are valid values for those fields. Eg.
```
                                 "maximumAttributeValueLength" : {
                                    "default" : "1024",
                                    "oneOf" : [
                                       {
                                          "maximum" : 2147483647,
                                          "minimum" : 0,
                                          "type" : "integer"
                                       },
                                       {
                                          "enum" : [
                                             "any"
                                          ],
                                          "type" : "string"
                                       },
                                       {
                                          "pattern" : "\\d+$",
                                          "type" : "string"
                                       }
                                    ]
                                 },
```

**Comment by fabriziofiorucci** [comment]
@schweits for some reason several fields apparently need to be set to integer. Applying an `APPolicy` manifest containing:

      defenseAttributes:
        allowCDATA: true
        allowDTDs: true
        allowExternalReferences: true
        allowProcessingInstructions: true
        maximumAttributeValueLength: any
        maximumAttributesPerElement: 32
        maximumChildrenPerElement: 4096
        maximumDocumentDepth: 32
        maximumDocumentSize: 1024000
        maximumElements: 512000
        maximumNSDeclarations: 64
        maximumNameLength: any
        maximumNamespaceLength: any
        tolerateCloseTagShorthand: true
        tolerateLeadingWhiteSpace: true
        tolerateNumericNames: true

Triggers:

```
The APPolicy "nginx-policy" is invalid: 
* spec.policy.xml-profiles[0].defenseAttributes.maximumAttributesPerElement: Invalid value: "integer": spec.policy.xml-profiles[0].defenseAttributes.maximumAttributesPerElement in body must be of type string: "integer"
* spec.policy.xml-profiles[0].defenseAttributes.maximumElements: Invalid value: "integer": spec.policy.xml-profiles[0].defenseAttributes.maximumElements in body must be of type string: "integer"
* spec.policy.xml-profiles[0].defenseAttributes.maximumDocumentSize: Invalid value: "integer": spec.policy.xml-profiles[0].defenseAttributes.maximumDocumentSize in body must be of type string: "integer"
* spec.policy.xml-profiles[0].defenseAttributes.maximumNSDeclarations: Invalid value: "integer": spec.policy.xml-profiles[0].defenseAttributes.maximumNSDeclarations in body must be of type string: "integer"
* spec.policy.xml-profiles[0].defenseAttributes.maximumDocumentDepth: Invalid value: "integer": spec.policy.xml-profiles[0].defenseAttributes.maximumDocumentDepth in body must be of type string: "integer"
* spec.policy.xml-profiles[0].defenseAttributes.maximumChildrenPerElement: Invalid value: "integer": spec.policy.xml-profiles[0].defenseAttributes.maximumChildrenPerElement in body must be of type string: "integer"
```

Unless I'm missing something, the manifest snippet should be correct according to https://docs.nginx.com/nginx-app-protect-waf/declarative-policy/policy/#policy/graphql-profiles/defenseAttributes

**Comment by schweits** [comment]
@fabriziofiorucci Right, integers should be valid, but according to the schema (and the doc you linked too), passing in a value like "1024000" should be valid too. But with the change here, wouldn't it reject strings no matter what?

**Comment by fabriziofiorucci** [comment]
@schweits as discussed, I'll check, test and add the definitive version of the CRD yaml shortly.

**Comment by galitskiy** [comment]
agree with @schweits on changing type to int; but not sure if "x-kubernetes-int-or-string: true" is a way to go either (I think there was a discussion long time ago with kic team, dont remember what we agreed on). @fabriziofiorucci also from your example of the manifest there are fields with the same type (int or string) set to "any".

**Comment by schweits** [comment]
@galitskiy I provided @fabriziofiorucci with the CRD we're using in our internal testing repo, so that's where many of those changes came from.

File: config/crd/bases/appprotect.f5.com_appolicies.yaml [hunk 1]
```diff
@@ -161,61 +174,114 @@ spec:
...
-                                  - VIOL_URL_LENGTH
-                                  - VIOL_URL_METACHAR
-                                  - VIOL_XML_FORMAT
-                                  - VIOL_XML_MALFORMED
+                                  - "VIOL_ACCESS_INVALID"
+                                  - "VIOL_ACCESS_MALFORMED"
+                                  - "VIOL_ACCESS_MISSING"
+                                  - "VIOL_ASM_COOKIE_HIJACKING"
+                                  - "VIOL_ASM_COOKIE_MODIFIED"
+                                  # - "VIOL_ATTACK_SIGNATURE"
```
Comment (vepatel) [inline config/crd/bases/appprotect.f5.com_appolicies.yaml:182]: @fabriziofiorucci any reason commented policy violations have been left in?

Comment (fabriziofiorucci) [reply config/crd/bases/appprotect.f5.com_appolicies.yaml:182]: @vepatel those are lines that came with the CRD used in the internal test repo, as provided by @schweits so I didn't change those

Comment (schweits) [reply config/crd/bases/appprotect.f5.com_appolicies.yaml:182]: Those are all valid values for that field, but the commented-out violations are for features that aren't supported by KIC yet. They can probably be removed here if they're not wanted

Comment (vepatel) [reply config/crd/bases/appprotect.f5.com_appolicies.yaml:182]: @fabriziofiorucci can you please remove them and rebase? I can approve the PR as well then 👍🏼

**Comment by vepatel** [comment]
@fabriziofiorucci Also the definitions in `deploy/crds-nap-waf.yaml` are generated automatically by running `make update-crds` from project root, so you only need to change `config/crds/bases`

**Review by vepatel — APPROVED** [review]
approved pending comment resolution

**Comment by fabriziofiorucci** [comment]
@vepatel for some reason `make update-crds` throws an error:

```$ make update-crds
go run sigs.k8s.io/controller-tools/cmd/controller-gen crd paths=./pkg/apis/... output:crd:artifacts:config=config/crd/bases
go: downloading sigs.k8s.io/controller-tools v0.13.0
[...]
go: finding golang.org/x/text v0.13.0
build sigs.k8s.io/controller-tools/cmd/controller-gen: cannot load io/fs: malformed module path "io/fs": missing dot in first path element
make: *** [Makefile:66: update-crds] Error 1
```

this happens on a Ubuntu 20.04 VM. Any clue?

**Review by shaun-nx — APPROVED** [review]

**Comment by shaun-nx** [comment]
Hi @fabriziofiorucci 
The PR is good to go now. If you rebase your branch I'll merge the PR as soon as the pipeline passes.

**Review by fabriziofiorucci — COMMENTED** [review]
reviewed

**Pull request merged**